packaging/windows/install_script.nsis

```diff
@@ -0,0 +1,278 @@
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "DisplayIcon" "$\"$INSTDIR\logo.ico$\""
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "Publisher" "$\"https://github.com/grafana/agent$\""
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "HelpLink" "$\"${HELPURL}$\""
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "URLUpdateInfo" "$\"${UPDATEURL}$\""
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "URLInfoAbout" "$\"${ABOUTURL}$\""
+    # There is no option for modifying or repairing the install
+    WriteRegDWORD HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "NoModify" 1
+    WriteRegDWORD HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "NoRepair" 1
+    Call WriteConfig
+
+   
+    # Create our batch file, since services cant run with parameters, nsexec is used to surpress console output, instead goes
+    # to NSIS log window
+    nsExec::ExecToLog 'sc create "Grafana Agent" binpath= "$INSTDIR\agent-windows-amd64.exe"'
+    Pop $0
+    # These separate create and config commands are needed, on the config the binpath is required
+    nsExec::ExecToLog 'sc config "Grafana Agent" binpath= "$INSTDIR\agent-windows-amd64.exe -config.file=\"$INSTDIR\agent-config.yaml\""'
+    Pop $0
+    nsExec::ExecToLog `sc start "Grafana Agent"`
+    Pop $0
+FunctionEnd
+
+
+Section "uninstall"
+    DetailPrint "Starting Uninstaller"
+    # Remove Start Menu launcher
+    delete "$SMPROGRAMS\${APPNAME}\${APPNAME}.lnk"
+    # Try to remove the Start Menu folder - this will only happen if it is empty
+    RMDir "$SMPROGRAMS\${APPNAME}"
+    # This is cleanup on the service and removing the exporter.
+    nsExec::ExecToLog `sc stop "Grafana Agent"`
+    Pop $0
+    nsExec::ExecToLog `sc delete "Grafana Agent" confirm`
+    Pop $0
+
+    # Remove files
+    delete $INSTDIR\agent-windows-amd64.exe
+    delete $INSTDIR\logo.ico
+    delete $INSTDIR\agent-config.yaml
+
+    # Always delete uninstaller as the last action
+    delete $INSTDIR\uninstall.exe
+
+    # Try to remove the install directory - this will only happen if it is empty
+    RMDir $INSTDIR
+
+    RMDir /r $APPDATA\grafana-agent-wal
+
+    # Remove uninstaller information from the registry
+    DeleteRegKey HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}"
+SectionEnd
+
+
+Function WriteConfig
+    # If the file exists we dont want to overwrite it since they may have custom code there
+    IfFileExists "$INSTDIR\agent-config.yaml" ReturnEarly WriteFile
+    ReturnEarly:
+        Return
+    WriteFile: 
+        # Write the config file, its easier to do this way than replacing an values in a templated file
+        FileOpen $9 "$INSTDIR\agent-config.yaml" w #Opens a Empty File and fills it
+        FileWrite $9 `server:$\n`
```

Comment:
Shouldn't these be `\r\n`?

Reply:
I vaguely recall having issues with \r\n when I did the first go around, but will try it out.